frontend/ui/otamanager.lua

```diff
@@ -481,10 +485,65 @@ function OTAManager:getOTAMenuTable()
                         text = _("Update channel"),
                         sub_item_table = self:genChannelList()
                     },
+                    {
+                        text_func = function()
+                            return T(_("Number of saved updates: %1"),
+                                self.nb_ota_packages ~= 0 and self.nb_ota_packages or _("unlimited"))
+                        end,
+                        keep_menu_open = true,
+                        callback = function(touchmenu_instance)
+                            UIManager:show(SpinWidget:new{
+                                    title_text = "Set number of saved downloaded updates",
+                                    info_text = _("Don't clean up, if number is set to zero."),
+                                    value = self.nb_ota_packages,
+                                    value_min = 0,
+                                    value_max = 99,
+                                    default_value = 3,
+                                    ok_text = _("Set"),
+                                    callback = function(spin)
+                                        self.nb_ota_packages = spin.value
+                                        G_reader_settings:saveSetting("nb_OTA_updates", spin.value)
+                                        self:cleanUpdates()
+                                        if touchmenu_instance then touchmenu_instance:updateItems() end
+                                    end,
+                                })
+                        end,
+                    },
                 }
             },
         }
     }
 end
 
+function OTAManager:cleanUpdates()
+    if self.nb_ota_packages <= 0 then
+        return
+    end
+
+    local update_template = "^koreader-" .. self:getOTAModel() .. "-.*-v.*"
```

Comment:
Regarding my comment about distinguishing stables and nightlies, if Version:getNormalizedCurrentVersion() returns a commit hash it's not a stable. Anyway, just a thought.